src/app/search/state/search.service.ts

```diff
@@ -108,20 +125,21 @@ export class SearchService {
    * @param {number} query_size
    * @memberof SearchService
    */
-  filterEntry(hits: Array<any>, query_size: number) {
+  filterEntry(hits: Array<Hit>, query_size: number): [Array<Hit>, Array<Hit>] {
     const workflowHits = [];
     const toolHits = [];
     hits.forEach(hit => {
       hit['_source'] = this.providerService.setUpProvider(hit['_source']);
       if (workflowHits.length + toolHits.length < query_size - 1) {
         if (hit['_type'] === 'tool') {
+          hit['_source'] = this.imageProviderService.setUpImageProvider(hit['_source']);
           toolHits.push(hit);
         } else if (hit['_type'] === 'workflow') {
           workflowHits.push(hit);
         }
       }
     });
-    this.setHits(toolHits, workflowHits);
+    return [toolHits, workflowHits];
   }
 
   setHits(toolHit: any, workflowHit: any) {
```

Comment:
Change this to take `Array<Hit>, Array<Hit>` as params.

Reply:
Changed from (toolHit: any, workflowHit:any) to (toolHit: Array<Hit>, workflowHit: Array<Hit>)